### DISCUSSION TITLE
same parameter，different PC,  different result

### DISCUSSION THREAD

**Discussion by SongLin99** [MaartenGr/BERTopic#2292] [Q&A] [answered]
Dear developer,
I run bertopic v0.16.4 on two windows11 pc, which have same version and parameters of HDBSCAN and UMAP.
and I saved the embeddings generated by SentenceTransformer funtion by following code because i think this could avoid random result of embeddings.

`embedding_model = SentenceTransformer("sentence-transformers/all-mpnet-base-v2", trust_remote_code=True)`
`embeddings = embedding_model.encode(abstract_list, batch_size=256, show_progress_bar=True, device='cuda')`
`np.save("10078_embed.npy", embeddings)`
`embeddings = np.load("10078_embed.npy")`

however, the topic generated on two computers is different
could you help me to solve this problem?

**parameters are here:**

`UMAP_neighbors = 15
UMAP_components = 20
UMAP_min_dist = 0.0
UMAP_metric = 'cosine'
HDB_size = 120
HDB_samples = 5
umap_model = UMAP(n_neighbors=UMAP_neighbors, n_components=UMAP_components, metric = UMAP_metric, min_dist=UMAP_min_dist, random_state=42)
hdbscan_model = HDBSCAN(min_cluster_size=HDB_size, min_samples=HDB_samples, metric='euclidean', cluster_selection_method='eom', prediction_data=True)

topic_model = BERTopic(
  embedding_model=embedding_model,
  umap_model=umap_model,
  hdbscan_model=hdbscan_model,
  ctfidf_model=ctfidf_model,
  vectorizer_model=vectorizer_model,

  top_n_words=10,
  verbose=True,
  calculate_probabilities=True,
  n_gram_range=(1,3),
)

topics, probs = topic_model.fit_transform(abstract_list, embeddings)`

**Comment by MaartenGr** [comment, marked as answer]
> I run bertopic v0.16.4 on two windows11 pc, which have same version and parameters of HDBSCAN and UMAP.
and I saved the embeddings generated by SentenceTransformer funtion by following code because i think this could avoid random result of embeddings.

Did you create the exact same environment on those two PCs? As in, did you create a python environment that has the exact same set of dependencies? The python versions need to be the same. Not just HDBSCAN and UMAP, but also numpy, scikit-learn, and any other package in the environment. That's what needed for any package/model if you want to perfectly reproduce the results.

**Reply by SongLin99** [reply]
Thank you，dear developer, now I know why the result is different, maybe the numpy version is different.
